atat/src/client.rs

```diff
@@ -176,12 +188,14 @@ where
                     }
                 })
                 .map_err(|e| {
-                    self.timer.try_start(self.config.cmd_cooldown).ok();
+                    self.timer
+                        .start(MillisDurationU32::from_ticks(self.config.cmd_cooldown).convert())
+                        .ok();
                     self.state = ClientState::Idle;
                     e
                 });
         } else if let Mode::Timeout = self.config.mode {
-            if self.timer.try_wait().is_ok() {
+            if self.timer.wait().is_ok() {
```

Comment:
This is not going to yield the same result..
This will block until the timer has run for `A::MAX_TIMEOUT`, whereas the `self.timer.try_wait()` will end up returning instantly no matter what, either with `Ok` or `WouldBlock`

Reply:
Ouch, yes you are right.

Reply:
So blocking wait is not what we want actually. Therefore new `Clock` trait must use `nb::Result` and `WouldBlock` as before.

Reply:
I am gonna improve `Clock` trait.

Reply:
Fixed.